app/models/jobstatusdata.rb

```diff
@@ -121,13 +123,27 @@ def extended_data_default(info)
     self.output_path = ''
 
     output_pathname = Pathname.new(ENV["HOME"])
-    self.terminal_path = OodAppkit.shell.url(path: (output_pathname.writable? ? output_pathname : ENV["HOME"]))
-    self.fs_path = OodAppkit.files.url(path: (output_pathname.writable? ? output_pathname : ENV["HOME"]))
+    self.file_explorer_url = { path: output_pathname }
+    self.shell_url = { path: output_pathname, cluster: self.cluster }
+
     self
   end
 
   private
 
+    def file_explorer_url=(path: path())
+      path = (path.writable? ? path : ENV["HOME"]).to_s
+
+      @file_explorer_url = OodAppkit.files.url(path: path).to_s
+    end
+
+    def shell_url=(path: path(), cluster: cluster())
+      path = (path.writable? ? path : ENV["HOME"]).to_s
+      host = OODClusters[cluster].login.host if OODClusters[cluster].login_allow?
+
+      @shell_url = host ? OodAppkit.shell.url(path: path, host: host).to_s : OodAppkit.shell.url(path: path).to_s
```

Comment:
You also could have gotten away with:

```ruby
opts = {}
opts[:path] = path.writable? ? path : ENV["HOME"]
opts[:host] = OODClusters[cluster].login.host if OODClusters[cluster].login_allow? && OODClusters[cluster].login.host

@shell_url = OodAppkit.shell.url(opts).to_s
```

although I should patch `#login_allow?` to return `false` if there is no `#login.host` set.